pandas/core/arrays/string_arrow.py

```diff
@@ -36,14 +36,20 @@
     except ImportError:
         pass
     else:
-        ARROW_CMP_FUNCS = {
-            "eq": pc.equal,
-            "ne": pc.not_equal,
-            "lt": pc.less,
-            "gt": pc.greater,
-            "le": pc.less_equal,
-            "ge": pc.greater_equal,
-        }
+        # pyarrow 0.16.0 adds a compute module (thus the above compute import
```

Comment:
rather than a try/except why don't we just simply check the version at this point. this nested try/except is very messy. or *way* better is just put a min of 1.0.0 on this functionaility.

Reply:
We discussed that on the original PR. The version check already happens elsewhere when actually trying to create an array (and *already* uses a minimum version of 1.0.0), so it's not needed to do a version check here.

Reply:
I remember, and obviously this is fragile. so this check needs to happen here as well.

Reply:
still -1 on this soln. i would just check that we are >= 1.0 and call it a day.